### PR TITLE
Pull tx data

### PR DESCRIPTION
**Pull request by infobiac**

### CODE REVIEW

File: BuyBitcoin/templates/list_transactions.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{% extends 'base.html' %}
+
+{% block content %}
+
+Transactions:<br>
+	{% for x in tx %}
+		{{x.name}}: ${{x.amount}}
+		<br>
```
Comment (Neitsch) [inline BuyBitcoin/templates/list_transactions.html]: Use `<p>{{x.name}}: ${{x.amount}}</p>` instead of <br>?

File: authentication/graphql.py [hunk 2]
```diff
@@ -32,7 +32,7 @@ class GProfile(DjangoObjectType):
     GraphQL representation of a Profile
     """
     stock_find = List(
-        GStock, args={'text': Argument(NonNull(String))})
+        GStock, args={'text': GStock, args={'text': Argument(NonNull(String))}})
```
Comment (Neitsch) [inline authentication/graphql.py]: Plz no

File: authentication/models.py [hunk 3]
```diff
@@ -42,14 +41,15 @@ def save_user_profile(instance, **_):
...
         User,
         on_delete=models.CASCADE
     )
     item_id = models.CharField(max_length=1000)
     access_token = models.CharField(max_length=1000)
+    institution_name = models.CharField(max_length=1000)
 
     def __str__(self):
-        return "{}, {}".format(self.id, self.user_id)
+        return "Institution: {}. IDs:{}, {}".format(self.institution_name, self.id, self.user_id)
```
Comment (Neitsch) [inline authentication/models.py]: id first please

File: authentication/views.py [hunk 4]
```diff
@@ -49,13 +49,37 @@ def get_access_token(request):
                               public_key=PLAID_PUBLIC_KEY, environment=PLAID_ENV)
         public_token = request.POST.get('public_token')
         exchange_response = client.Item.public_token.exchange(public_token)
+        r = client.Item.get(exchange_response['access_token'])
         bank_user = UserBank(
             user=request.user,
             item_id=exchange_response['item_id'],
-            access_token=exchange_response['access_token']
+            access_token=exchange_response['access_token'],
+            institution_name=r['item']['institution_id']
```
Comment (Neitsch) [inline authentication/views.py]: trailing comma

File: authentication/views.py [hunk 5]
```diff
@@ -49,13 +49,37 @@ def get_access_token(request):
...
             )
```
Comment (Neitsch) [inline authentication/views.py]: Is the indentation right?

File: authentication/views.py [hunk 6]
```diff
@@ -49,13 +49,37 @@ def get_access_token(request):
...
+    """
+    A demo view to list all of a user's transactions
+    """
+    if request.method=="POST":
+        user_bank_id = request.POST.get('id')
+        user_bank = UserBank.objects.get(pk=user_bank_id)
+        if user_bank.user == request.user:
+            client = plaid.Client(client_id=PLAID_CLIENT_ID, secret=PLAID_SECRET,
+            	public_key=PLAID_PUBLIC_KEY, environment=PLAID_ENV)
+            response = client.Transactions.get(user_bank.access_token, start_date='2016-07-12', end_date='2017-01-09')
```
Comment (Neitsch) [inline authentication/views.py]: Hardcoded dates?

File: authentication/views.py [hunk 7]
```diff
@@ -49,13 +49,37 @@ def get_access_token(request):
...
+            transactions = response['transactions']
+            c = {'tx': transactions}
+            return render(request, "list_transactions.html", c)
+        else:
+        	return HttpResponse("You don't have permission to view that.")
+    else:
+    	userbank = request.user.userbank_set.all()
+    	c = {'ub': userbank}
```
Comment (Neitsch) [inline authentication/views.py]: userbank instead of ub for clarity on the client side

File: authentication/models.py [hunk 8]
```diff
@@ -42,14 +41,15 @@ def save_user_profile(instance, **_):
...
-    Contains all the user's bank access data
+    Contains all the user's bank access data (via plaid)
     """
     user = models.ForeignKey(
         User,
         on_delete=models.CASCADE
     )
     item_id = models.CharField(max_length=1000)
     access_token = models.CharField(max_length=1000)
+    institution_name = models.CharField(max_length=1000)
```
Comment (Neitsch) [inline authentication/models.py:53]: That's hella long. We can probably cut all the char lengths down. Plaid should provide specs?

File: BuyBitcoin/templates/bank_select_form.html [hunk 9]
```diff
@@ -0,0 +1,14 @@
+{% extends 'base.html' %}
+
+{% block content %}
+
+Select a bank:
+	{% for x in ub %}
```
Comment (Neitsch) [inline BuyBitcoin/templates/bank_select_form.html]: plz betternaming

File: BuyBitcoin/templates/list_transactions.html [hunk 10]
```diff
@@ -0,0 +1,10 @@
+{% extends 'base.html' %}
+
+{% block content %}
+
+Transactions:<br>
+	{% for x in tx %}
```
Comment (Neitsch) [inline BuyBitcoin/templates/list_transactions.html]: plz better naming

**Review by Neitsch — COMMENTED** [review]
Could you add screenshots?

File: authentication/models.py (hunk 8 above)
Comment (infobiac) [reply authentication/models.py:53]: I don't think there's any performance difference in terms of postgresql; this is just (required) validation. I checked plaid and unfortunately there's nothing there.

**Comment by infobiac** [comment]
Screenshots: https://imgur.com/a/cKWrP  The bank selection screen only has "tb" for now because i named the two the same thing when creating the DB.

File: authentication/views.py [hunk 11]
```diff
@@ -49,13 +49,37 @@ def get_access_token(request):
...
+    A demo view to list all of a user's transactions
+    """
+    if request.method == "POST":
+        user_bank_id = request.POST.get('id')
+        user_bank = UserBank.objects.get(pk=user_bank_id)
+        if user_bank.user == request.user:
+            client = plaid.Client(client_id=PLAID_CLIENT_ID, secret=PLAID_SECRET,
+                                  public_key=PLAID_PUBLIC_KEY, environment=PLAID_ENV)
+            response = client.Transactions.get(user_bank.access_token,
+                                               start_date='2016-07-12', end_date='2017-01-09')
```
Comment (Neitsch) [inline authentication/views.py:78]: Hardcoded dates?

**Review by Neitsch — APPROVED** [review]
That's a lot of money spent on KFC

File: authentication/views.py (hunk 11 above)
Comment (infobiac) [reply authentication/views.py:78]: yea, i gotta fix that. This was just pulling initial tx data; I'll be switching to yearly from present day next.

**Pull request merged**